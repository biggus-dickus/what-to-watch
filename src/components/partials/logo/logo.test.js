import React from 'react';
import {StaticRouter as Router} from 'react-router-dom';

import {configure, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import RouteConfig from '../../../config/routes';
import Logo from './logo';
import Adapter from "enzyme-adapter-react-16/build";


configure({adapter: new Adapter()});

describe(`Logo test suite`, () => {
  it(`<Logo /> should render correctly`, () => {
    const tree = renderer.create(
        <Router><Logo pathname={RouteConfig.SIGN_IN} /></Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`should contain a link to the home page if it's rendered by any route except home`, () => {
    const wrapper = mount(<Router><Logo pathname={RouteConfig.SIGN_IN} /></Router>);
    const link = wrapper.find(`Link`);

    expect(link).toHaveLength(1);
    expect(link.prop(`to`)).toEqual(RouteConfig.INDEX);

    wrapper.unmount();
  });

  it(`should NOT contain a link if it's rendered on home page per se`, () => {
    const wrapper = mount(<Router><Logo pathname={RouteConfig.INDEX} /></Router>);
    const link = wrapper.find(`Link`);

    expect(link).toHaveLength(0);

    wrapper.unmount();
  });
});

