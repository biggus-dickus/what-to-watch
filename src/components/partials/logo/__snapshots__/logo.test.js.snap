// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render correctly 1`] = `
<div
  className="logo"
>
  <a
    className="logo__link"
    href="/"
    onClick={[Function]}
  >
    <span
      className="logo__letter logo__letter--1"
    >
      W
    </span>
    <span
      className="logo__letter logo__letter--2"
    >
      T
    </span>
    <span
      className="logo__letter logo__letter--3"
    >
      W
    </span>
  </a>
</div>
`;

exports[`Logo Block test suite <Logo /> should render correctly 1`] = `
<div
  className="logo"
>
  <a
    className="logo__link"
    href="/"
    onClick={[Function]}
  >
    <span
      className="logo__letter logo__letter--1"
    >
      W
    </span>
    <span
      className="logo__letter logo__letter--2"
    >
      T
    </span>
    <span
      className="logo__letter logo__letter--3"
    >
      W
    </span>
  </a>
</div>
`;

exports[`Logo test suite <Logo /> should render correctly 1`] = `
<div
  className="logo"
>
  <a
    className="logo__link"
    href="/"
    onClick={[Function]}
  >
    <span
      className="logo__letter logo__letter--1"
    >
      W
    </span>
    <span
      className="logo__letter logo__letter--2"
    >
      T
    </span>
    <span
      className="logo__letter logo__letter--3"
    >
      W
    </span>
  </a>
</div>
`;
